src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -27,6 +28,559 @@
 
 using namespace Microsoft::Console::Render;
 
+namespace
+{
+    // From: https://learn.microsoft.com/en-us/windows/win32/direct3d11/overviews-direct3d-11-resources-textures-how-to

```

Comment:
This code might be outdated. It's actually taken from here which is also the code we might want to copy if anything: <https://github.com/microsoft/DirectXTK/blob/2530d168f50ecca48bf4f2f7f22a59f840939816/Src/WICTextureLoader.cpp>

I noticed that you renamed/merged some functions which I'd prefer if that wasn't the case, if I'm honest. Simply because those original function names will make finding and comparing this code with the original source of this code easier.

Reply:
In progress...